mirrord-agent/src/file.rs

```diff
@@ -0,0 +1,198 @@
+
+                ReadFileResponse {
+                    bytes: buffer,
+                    read_amount,
+                }
+            })
+            .map_err(|fail| {
+                ResponseError::FileOperation(FileError {
+                    operation: format!("read"),
+                    raw_os_error: fail.raw_os_error(),
+                    kind: fail.kind().into(),
+                })
+            })
+    }
+
+    pub(crate) fn seek(
+        &mut self,
+        fd: i32,
+        seek_from: SeekFrom,
+    ) -> Result<SeekFileResponse, ResponseError> {
+        let file = self.open_files.get_mut(&fd).unwrap();
+        debug!("FileManager::seek -> Trying to seek file {file:#?}, with seek {seek_from:#?}");
+
+        file.seek(seek_from)
+            .map(|result_offset| SeekFileResponse { result_offset })
+            .map_err(|fail| {
+                ResponseError::FileOperation(FileError {
+                    operation: format!("seek"),
+                    raw_os_error: fail.raw_os_error(),
+                    kind: fail.kind().into(),
+                })
+            })
+    }
+
+    pub(crate) fn write(
+        &mut self,
+        fd: i32,
+        write_bytes: Vec<u8>,
+    ) -> Result<WriteFileResponse, ResponseError> {
+        let file = self.open_files.get_mut(&fd).unwrap();
+        debug!("FileManager::write -> Trying to write file {file:#?}");
+
+        file.write(&write_bytes)
+            .map(|written_amount| WriteFileResponse { written_amount })
+            .map_err(|fail| {
+                ResponseError::FileOperation(FileError {
+                    operation: format!("write"),
+                    raw_os_error: fail.raw_os_error(),
+                    kind: fail.kind().into(),
+                })
+            })
+    }
+
+    pub(crate) fn close(&mut self, fd: i32) -> Result<CloseFileResponse, ResponseError> {
+        let file = self.open_files.remove(&fd).unwrap();
+        debug!("FileManager::write -> Trying to close file {file:#?}");
+
+        Ok(CloseFileResponse)
+    }
+}
+
+pub async fn file_worker(
+    mut file_request_rx: Receiver<(PeerID, FileRequest)>,
+    file_response_tx: Sender<(PeerID, FileResponse)>,
+    container_id: Option<String>,
+    file_manager: Arc<Mutex<FileManager>>,
+) -> Result<(), AgentError> {
+    debug!("file_worker -> Setting namespace");
+
+    let pid = match container_id {
+        Some(container_id) => Runtime::get_container_pid(&container_id, "containerd").await,
+        None => Err(AgentError::NotFound("Empty container_id!".to_string())),
+    }?;
+
+    let root_path = PathBuf::from("/proc").join(pid).join("root");
+    debug!("file_worker -> root path {:#?}", root_path);
+
+    // TODO: To open the file we need access to the `FileManager`, should it be wrapped in
+    // `Arc<Mutex>>` and passed here via `clone`? Can we keep the `FileManager` living here
+    // forever (create it in this thread)?
```

Comment:
I don't see why FileManager shuold be Arc/Mutex as it's only used in this function?